redpanda/resources/resourcegroup/resource_resourcegroup.go

```diff
@@ -0,0 +1,182 @@
+	"github.com/redpanda-data/terraform-provider-redpanda/redpanda/models"
+	"github.com/redpanda-data/terraform-provider-redpanda/redpanda/utils"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var (
+	_ resource.Resource                = &ResourceGroup{}
+	_ resource.ResourceWithConfigure   = &ResourceGroup{}
+	_ resource.ResourceWithImportState = &ResourceGroup{}
+)
+
+// ResourceGroup represents a cluster managed resource.
+type ResourceGroup struct {
+	CpCl *cloud.ControlPlaneClientSet
+}
+
+// Metadata returns the full name of the ResourceGroup resource.
+func (*ResourceGroup) Metadata(_ context.Context, _ resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "redpanda_resourcegroup"
+}
+
+// Configure uses provider level data to configure ResourceGroup client.
+func (n *ResourceGroup) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	if request.ProviderData == nil {
+		// We can't add a diagnostic for an unset ProviderData here because
+		// during the early part of the terraform lifecycle, the provider data
+		// is not set and this is valid, but we also can't do anything until it
+		// is set.
+		return
+	}
+
+	p, ok := request.ProviderData.(config.Resource)
+	if !ok {
+		response.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *provider.Data, got: %T. Please report this issue to the provider developers.", request.ProviderData),
+		)
+		return
+	}
+	n.CpCl = cloud.NewControlPlaneClientSet(p.ControlPlaneConnection)
+}
+
+// Schema returns the schema for the ResourceGroup resource.
+func (*ResourceGroup) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = resourceGroupSchema()
+}
+
+// resourceGroupSchema defines the schema for a resource group.
+func resourceGroupSchema() schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Required:      true,
+				Description:   "Name of the resource group. Changing the name of a resource group will result in a new resource group being created and the old one being destroyed",
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+			"id": schema.StringAttribute{
+				Computed:      true,
+				Description:   "UUID of the resource group",
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+		},
+		Description: "A Redpanda Cloud resource group",
+		Version:     1,
+	}
+}
+
+// Create creates a new ResourceGroup resource. It updates the state if the
+// resource is successfully created.
+func (n *ResourceGroup) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var model models.ResourceGroup
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &model)...)
+
+	rg, err := n.CpCl.CreateResourceGroup(ctx, model.Name.ValueString())
```

Comment:
would this function call fail if the resource group already exist? can we make it idempotent?

Reply:
No, there are multiple scenarios:

1. If you manually create the resource group and attempt to create one with the same name, **it will fail saying that the resource already exists**. The proper way to handle this is to import the state or use the namespace as a data source. ([Terraform Import](https://developer.hashicorp.com/terraform/cli/import), [Terraform Data Source](https://developer.hashicorp.com/terraform/language/data-sources), we provide the resource group as data source in the file data_resourcegroup.go)
2. You run terraform apply, create the resource group, and attempt to run terraform apply. Nothing will happen, there is no resource to change.